docs/maven/build-integration.md

```diff
@@ -0,0 +1,192 @@
+# Build Integration
+
+[[toc]]
+
+## Introduction
+
+Integrating the jSparrow Maven plugin into a build process has the advantage of automatically applying selected jSparrow rules in an already automated process. 
+The configuration is highly customizable and depends very much on the environment. 
+Here it will be described how Jenkins builds are able to trigger jSparrow Maven plugin builds, depending on the Git branch of the commit. In this example, all builds on a feature branch will trigger jSparrow. 
+
+## Requirements
+
+This description assumes that the following components are available and correctly setup. 
+
+* [Jenkins](https://www.jenkins.io/download/)
+  * [Pipeline Maven Integration plugin](https://plugins.jenkins.io/pipeline-maven/)
+  * [Bitbucket Server Notifier plugin](https://github.com/jenkinsci/stashnotifier-plugin/blob/release/1.x/readme.md)
+  * [SSH Agent plugin](https://plugins.jenkins.io/ssh-agent/)
+  * ([Declarative](https://www.jenkins.io/doc/book/pipeline/syntax/#declarative-pipeline)) [Multibranch Pipeline](https://www.jenkins.io/doc/book/pipeline/multibranch/)
+* [Bitbucket](https://hub.docker.com/r/atlassian/bitbucket-server/)
+
+## Jenkins Configuration
+
+A jSparrow build requires a license key and a config file (jsparrow.yml) to be present. Both can be kept securely within Jenkins and then referenced by the pipeline. 
```

Comment:
Question: 
Is it possible to have a config.yml file that contains the license key and the license server URL, similar to what we have in our development machines? This would simplify the configuration.

Reply:
Yes, it is possible. It could be done just like it is done with the jsparrow.yml file. It made sense to me to keep using a secret text for the license because that way both options are explained.

Reply:
Perfect. Yes yes. I also agree. We should just keep in mind that we can do the config.yml file.